crates/utils/sync/src/racy_lock.rs

```diff
@@ -26,6 +26,97 @@ where
     f: F,
 }
 
+#[cfg(all(loom, test))]
+mod unsound_demo {
+    use alloc::boxed::Box;
+    use core::{
+        cell::RefCell,
+        ptr,
+        sync::atomic::{AtomicPtr, Ordering},
+    };
+
+    use loom::{hint, model::Builder, sync::Arc, thread};
+
+    // Deliberately unsound lock that ignores `T` in Sync/Send bounds to demonstrate the failure.
```

Comment:
Nice touch adding the `unsound_demo` module! This really helps illustrate why the bounds are needed. The loom test with `RefCell` clearly shows the runtime panic that can happen without proper bounds.